de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerViewController.java

```diff
@@ -341,15 +340,13 @@ public void toggleView(@UIEventTopic(E4CEventTable.EVENT_VIEW_TOGGLE) Object o)
 		services.eventBroker.send(E4CEventTable.EVENT_REFRESH_PROJECT_VIEWER, null);
 	}
 
-	/**
-	 * Subscribing on the drop element event
-	 */
+	/** Subscribing on the drop element event */
 	@Inject
 	@Optional
 	public void dropElements(@UIEventTopic(E4CEventTable.EVENT_DROP_ELEMENT_IN_EXPLORER) Object o) {
 		if (o instanceof DropElement) {
-			WizardDialog dialog = new WizardDialog(_shell, new DropWizard(context, (DropElement) o, imageService));
-			dialog.open();
+			DropFilesDialog dialog = new DropFilesDialog(context, (DropElement) o, imageService);
+			dialog.showDialog();
 		}
 	}
 

```

Comment:
Handle SET_SELECTION event here

Reply:
As you said, the event is unused - it's a remnant from some testing done  with the NewFolderHandler

Reply:
But it should be used to prevent the current inconsistent UI behavior at least

Reply:
this behaviour of reseting the visual selection, not the selection in the service is consistent, as far as i can tell, on every action that changes the working tree and forces a refresh. Maybe put that in an independant task?